tools/integration-tests/tester/tests/testutil.go

```diff
@@ -164,6 +179,49 @@ func CheckForMessageIDs(t *testing.T, peers []*framework.Peer, messageIDs map[st
 	}
 }
 
+// AwaitMessageAvailability awaits until the given message IDs become available on all given peers or
+// the max duration is reached. Returns a map of missing messages per peer. An error is returned if at least
+// one peer does not have all specified messages available.
+func AwaitMessageAvailability(peers []*framework.Peer, messageIDs map[string]DataMessageSent, waitFor time.Duration) (missing map[identity.ID]map[string]types.Empty, err error) {
+	s := time.Now()
+	missing = map[identity.ID]map[string]types.Empty{}
```

Comment:
Need a mutex to protect the concurrent read/write below.
I just noticed that :stuck_out_tongue:

Reply:
Do we really need the lock? All threads operate on separate elements of the map. If so we only need it for checking the first level, ie `map[identity.ID]` I think

Reply:
I've added the lock and the other optimization you suggested as well

Reply:
I deleted the other comment simply because it will skip the check in the first round, and I didn't come up with other solution.
But just add the additional check on `i` will work, just realize that from your fix! :clinking_glasses: